test_sqlalchemy.py

```diff
@@ -456,6 +456,44 @@ class Baz(db.Model):
             Baz.__table__: db.get_engine(app, None)
         })
 
+    def test_abstract_binds(self):
+        _, db1 = tempfile.mkstemp()
+        _, db2 = tempfile.mkstemp()
+
+        def _remove_files():
+            try:
+                os.remove(db1)
+                os.remove(db2)
+            except IOError:
+                pass
+        atexit.register(_remove_files)
+
+        app = flask.Flask(__name__)
+        app.config['SQLALCHEMY_ENGINE'] = 'sqlite://'
+        app.config['SQLALCHEMY_BINDS'] = {
+            'foo':      'sqlite:///' + db1,
+            'bar':      'sqlite:///' + db2
+        }
+        db = sqlalchemy.SQLAlchemy(app)
+
+        class AbstractFooBoundModel(db.Model):
+            __abstract__ = True
+            __bind_key__ = 'foo'
+
+        class FooBoundModel(AbstractFooBoundModel):
+            id = db.Column(db.Integer, primary_key=True)
+
+        db.create_all()
+
+        # does the model have the correct engines?
+        self.assertEqual(db.metadata.tables['foo_bound_model'].info['bind_key'], 'foo')
+
+        # see the tables created in an engine
```

Comment:
This section seems unnecessary.  Could be replaced with `db.get_engine(bind='foo').execute('select * from users')`, which would raise if the table didn't exist, but we're not really testing that binds work here.